### ISSUE TITLE
MAJOR FLAW when replacing from bsx

### ISSUE THREAD

**Issue by prakash-niroula** [face-hh/bussin#8]
Consider this
```bsx
lit x be "Hey do you want this ? fr fr !" rn
waffle(x)
```

Expected : Hey do you want this ? fr fr !
Got : Hey do you want this ? == == !

**Comment by Polygons1** [comment]
true

**Comment by face-hh** [comment]
Interesting, it must be a glitch inside the Regular Expression:

```ts
String.prototype.replace_fr = function (target: string, replacement: string): string {
    const pattern = new RegExp('(?<![\'"`])\\b' + target + '\\b(?!["\'`])', 'g');
    
    return this.replace(pattern, replacement);
}```

It's supposed to replace keywords that are **not** inside strings, and it works well for words like "lit".

Maybe someone who's more experienced in RegExp can look into this?

**Comment by prakash-niroula** [comment]
> Interesting, it must be a glitch inside the Regular Expression:
> 
> ```ts
> String.prototype.replace_fr = function (target: string, replacement: string): string {
>     const pattern = new RegExp('(?<![\'"`])\\b' + target + '\\b(?!["\'`])', 'g');
>     
>     return this.replace(pattern, replacement);
> }
> ```
> 
> It's supposed to replace keywords that are **not** inside strings, and it works well for words like "lit".
 
> Maybe someone who's more experienced in RegExp can look into this?

yeah it's because of regexp
![image](https://github.com/face-hh/bussin/assets/82296809/51fd3805-5e84-47d9-8e1b-f0f00208da29)

**Comment by prakash-niroula** [comment]
I think it's because of space. " fr " is a match but "fr" isn't

**Comment by prakash-niroula** [comment]
> It's supposed to replace keywords that are **not** inside strings, and it works well for words like "lit".
@face-hh It doesn't
```bsx
waffle("Regex check")
lit word be "hey lit x be 10 fr 10 rn " rn
waffle(word)
```

Produces

```
hey let x = 10 == 10 ;
```

**Comment by Symmettry** [comment]
fixed in #30

**Comment by face-hh** [comment]
Fixed in #30
